### PR TITLE
UI-492 BreadCrumbs

### PR DESCRIPTION
**Pull request by hanzge**
- BreadCrumb component refactor
- Move Crumb to another file
- Remove `href` from Crumb
- BreadCrumb storybook update]

@mallik-svmx @sheela-svmx @joshsylvester @kokasvmx 

The reason to remove `href` from Crumb is that clicking `<a href="/xxx">link</a>` will jump to and reload the page completely. It's not what single page app should be like. In my implementation, developer should write link element themselves rather than pass href to props.

Something like this:
```
import { Link } from 'react-router';

<BreadCrumbs>
  <Crumb>
    <Link to="/home">Home</Link>
  </Crumb>
  <Crumb>
    <Link to="/home">Products</Link>
  </Crumb>
</BreadCrumbs>
```

### CODE REVIEW

File: lib/components/BreadCrumbs/BreadCrumbs.js [hunk 1]
```diff
@@ -2,47 +2,38 @@ import React from 'react';
...
+   * Extra className besides default ones
+   */
+  className: PropTypes.string,
 };
 
-Crumb.propTypes = {
-  href: PropTypes.string,
-  className: PropTypes.string,
+const defaultProps = {
   children: PropTypes.node,
```
Comment (mallik-svmx) [inline lib/components/BreadCrumbs/BreadCrumbs.js]: Update defaultProps

Comment (hanzge) [reply lib/components/BreadCrumbs/BreadCrumbs.js]: Fixed.

File: lib/components/BreadCrumbs/Crumb.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+  className: PropTypes.string,
+};
+
+const defaultProps = {
+  children: null,
+  className: '',
+};
+
+const Crumb = ({ className, children, ...restProps }) => {
+  const CrumbClassName = classnames('Crumb slds-list__item slds-text-heading--label', className);
```
Comment (joshsylvester) [inline lib/components/BreadCrumbs/Crumb.js]: this should be camelCased as it's not a component. `crumbClassName`

File: lib/components/BreadCrumbs/BreadCrumbs.js [hunk 3]
```diff
@@ -2,47 +2,38 @@ import React from 'react';
...
+const defaultProps = {
+  children: null,
+  className: '',
 };
 
-const BreadCrumbs = ({ label, className, children, ...props }) => {
-  const oClassName = classnames(
-    'slds-breadcrumb slds-list--horizontal',
+const BreadCrumbs = ({ className, children, ...restProps }) => {
+  const BreadCrumbsClassName = classnames(
```
Comment (joshsylvester) [inline lib/components/BreadCrumbs/BreadCrumbs.js]: should be camelCased as it's not a component. `breadCrumbsClassName`

File: lib/components/BreadCrumbs/Crumb.test.js [hunk 4]
```diff
@@ -0,0 +1,11 @@
+import React from 'react';
+import { shallow } from 'enzyme';
+
+import Crumb from './Crumb';
+
+describe('Crumb', () => {
+  it('should have classNames', () => {
+    const wrapper = shallow(<Crumb />);
+    expect(wrapper.prop('className')).toEqual('Crumb slds-list__item slds-text-heading--label');
```
Comment (joshsylvester) [inline lib/components/BreadCrumbs/Crumb.test.js]: this test is likely to fail further down the line if the component gets updated. 

i'd suggest that we update it to check for the classes that we care about, with an expect for `hasClass`

**Comment by hanzge** [comment]
@joshsylvester Fixed.

**Comment by joshsylvester** [comment]
two approvals, merging

**Pull request merged**